### PR TITLE
order description and stats

### LINKED ISSUE #40: Calculate Order stats
Currently we have some place holders to show order area and amount, fix them to show the actual values in invoice table.

### LINKED ISSUE #53: Input Number icons not working.
The Numberinput fields for price and quantity, the icons for high and low don't work.

### LINKED ISSUE #54: Order description is not editable.
In Invoice table, the order description field is not editable, please take a look at that. 

Also do the order stats in that row, I think the current PR can be rejected.

### LINKED ISSUE #56: Edit invoice.
When editing an invoice, following are possible actions and how we want to handle them. 

- Delete should immediately delete row (Order or item)
- Add a new Item/Order  -->  item.state === 'added'
- Update existing Item/Order  -->  item.state === 'modified'

For inspiration about how to structure the code, take a look at invoice create.jsx

### PR DESCRIPTION
**Pull request by SalmanMughal789**
I am excited to tell you that this pull request resolves the following issues of orders of invoice.

- The order description is auto-generated

- The order description is editable

- The Stats of totalAmount and totalArea of orders are displayed and updates in real time as the user changes anything of order items improving the overall user experience. 

- The inputNumber bug is addressed. Price and quantity does not accept string and the controls to increase or decrease the value can be controlled by the buttons or from keyboard. 

- There is one generic method which updates the stats whenever any changes occurs in image-area-price-height-width-quantity  etc

Closed #53 
Closed #54 
Closed #40 
Closed #56 

@Waqar-SE Please review

### CODE REVIEW

File: frontend/context/BillContext.jsx [hunk 1]
```diff
@@ -208,10 +224,16 @@ export function BillProvider({ children }) {
...
-
         if (!action.payload.orderId && action.payload.itemId) {
-          newState.selectedInvoice.items = state.selectedInvoice?.items.map(item =>
-            item.id === action.payload.itemId ? updateRow(item, action.payload) : item)
+          const isItem = newState.selectedInvoice.items.find(item => item.id === action.payload.itemId);
+          if(isItem){
+            newState.selectedInvoice.items = state.selectedInvoice?.items.map(item =>
+              item.id === action.payload.itemId ? updateRow(item, action.payload) : item)
+          }
+          else{
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx:233]: It's ambigious to handle orders when orderId is null/false, please change it.

**Review by Waqar-SE — CHANGES_REQUESTED** [review]

File: frontend/context/BillContext.jsx (hunk 1 above)
Comment (Waqar-SE) [reply frontend/context/BillContext.jsx:233]: If this is to cater description, please add a separate handler for description of order instead.

File: frontend/context/BillContext.jsx [hunk 2]
```diff
@@ -203,26 +219,56 @@ export function BillProvider({ children }) {
         break;
 
       case "setDescription":
+
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx]: We can't rely on updateRow function here as that reset the amount and area for Row which should not be updated when the description is changed. 

Instead change the description directly by doing something like 
`({ ...item, description: action.payload.description,state:'modified' })` in place of updateRow function call.

File: frontend/context/BillContext.jsx [hunk 3]
```diff
@@ -203,26 +219,56 @@ export function BillProvider({ children }) {
...
+      }
+      else {
+        const orderIndex = getOrderIndex(state, action.payload.orderId);
+        if (orderIndex !== -1) {
+          newState.selectedInvoice.orders[orderIndex].items = state.selectedInvoice.orders[orderIndex].items.map(item =>
+            item.id === action.payload.itemId
+              ? updateRow(item, action.payload)
+              : item)
+        }
+        else {
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx:232]: We don't need this else branch as this same case is handled in the first condition where you're checking for itemId and item model.

File: frontend/components/invoices/Table.jsx [hunk 4]
```diff
@@ -117,7 +117,7 @@ function Table({ title, invoiceId = null, TableActions }) {
       dataIndex: "area",
       key: "area",
       align: "center",
-      render: (text, row) => isOrderRow(row) ? 'Order Area' : Number(text).toFixed(2)
+      render: (text, row) => isOrderRow(row) ? row.area ? row.area : 'Order Area' : Number(text).toFixed(2)
```
Comment (Waqar-SE) [inline frontend/components/invoices/Table.jsx:120]: Don't fall back to "Order area" but show the area without checking if it's null or not. Currently it hides incorrect calculations. 

To make sure you don't show NAN of undefined you can do `row.area || 0'

File: frontend/components/invoices/Table.jsx [hunk 5]
```diff
@@ -151,7 +152,7 @@ function Table({ title, invoiceId = null, TableActions }) {
       dataIndex: "amount",
       key: "amount",
       align: "center",
-      render: (value, row) => isOrderRow(row) ? 'Order Total' : Number(value?.toFixed(2))
+      render: (value, row) => isOrderRow(row) ? row.amount? row.amount : 'Order Total' : Number(value?.toFixed(2))
```
Comment (Waqar-SE) [inline frontend/components/invoices/Table.jsx:155]: Do the same thing here as order area cell.

**Review by Waqar-SE — CHANGES_REQUESTED** [review]
Changes required. Also the order stats don't show up when we open an invoice in edit mode. Please fix that as well.

**Comment by SalmanMughal789** [comment]
> Changes required. Also the order stats don't show up when we open an invoice in edit mode. Please fix that as well.

Its done

File: frontend/context/BillContext.jsx [hunk 6]
```diff
@@ -84,16 +102,20 @@ export function BillProvider({ children }) {
...
+          payload: data
+        })
+
     } catch (error) {
       console.error('Failed to fetch invoices:', error);
     }
   };
 
   const handleSaveInvoice = async () => {
+    debugger;
```
Comment (Waqar-SE) [inline frontend/context/BillContext.jsx:117]: Remove this statement.

**Review by Waqar-SE — CHANGES_REQUESTED** [review]

**Review by Waqar-SE — APPROVED** [review]

**Pull request merged**